### ISSUE TITLE
Javascript: performance and latency issues

### ISSUE THREAD

**Issue by toconnell** [toconnell/kdm-manager#269]
From email:

> [5961b1908740d969d5fd6a08] has submitted an error report!
> 
> The report goes as follows:
> TOC, 
> 
> Hello, I am sending this email in order to bring attention to a possible issue in the Settlement sheet.
> I've noticed that the option to decrease population often doesn't respond.
> I continually find myself having to manually input my settlement's correct population. I find that it takes quite awhile now for the manager to save information as well. 

This could be a consequence of bad JS, e.g. the sleep-like stuff that happens after certain operations or the reinitialize() method causing chaos on the sheet generally. 

Either way, this is worth checking out.

**Comment by toconnell** [comment]
@wayforger Just emailed you about this.

**Comment by toconnell** [comment]
OK, so, I just spent a few hours with this over the last couple of days, and it's looking like this was consequence of loading the settlement several times during Settlement Sheet initialization via the apiService factory.

I ended up fixing a lot of this in the course of resolving this issue: https://github.com/toconnell/kdm-manager/issues/275

Basically, resolving #275 makes it so that the settlement sheet does a single API call to pull the settlement down and then initializes all page stuff (except for Lost Settlements: we'll have to fix that later) in one shot.

The problem here was that the old initialize() method loaded the settlement several times, and the normal methods in the controller were getting stepped on by asynchronous initialize() and reinitialize() calls.

tl;dr this should be solved in the next release, so I'm marking it closed for now.
